### ISSUE TITLE
This repo has to be updated.

### ISSUE THREAD

**Issue by lillallol** [Acmion/SvelteScaling#2]
Everything said in [this](https://github.com/halfnelson/svelte-it-will-scale/issues/3) issue are valid for this repo as well. Please consider updating it. I am looking forward to it.

**Comment by Acmion** [comment]
Hmm... Treating the intercept as a known value may not necessarily be a good approach. This is because the frameworks themselves could, at least in theory, utilize tree shaking.

**Comment by lillallol** [comment]
@Acmion If `y` for `x = 0` is equal or less than the bundle size of the framework, then yeah tree shaking is happening. Is that indeed the case for what you have done? 

By the way have you made your benchmark explicitly use a specific version of svelte? If yes, then which version is it?

Edit: Also lets not forget that we are interested for a **general** equation about the interception point of svelte. That means we are not interested for **specific** cases. Hence it makes no sense to use the tree shaked intercept since what is tree shaked for a specific app might not be tree shaked for another.

**Comment by Acmion** [comment]
The gzipped size of the app with no components (see SvelteScaling/sample-apps/svelte-apps/svelte-app-0/) in this study is 1344 bytes. However, I would not take this to be representative of the true bundle size of the framework.

The version of Svelte in the tests is 3.24.1. This can be found out from the `package-json.lock` files. I could document this better. For reference, the newest Svelte version is currently 3.48.0.

Yes, we are interested in a general equation, but I do not think that there is a specific "known" value for the intercept. For example, the study that you linked in the other repo (https://github.com/yyx990803/vue-svelte-size-analysis#analysis) does pretty much the same analysis as in this repo, except only for a single component. 

The components in this repo use a mix and match of features and the gzipped compiled size correspond quite well to the linear regression, which would indicate that the intercept (or bundle size of the framework) is not too far from the reported value.

**Comment by lillallol** [comment]
> which would indicate that the intercept (or bundle size of the framework) is not too far from the reported value.

You have to change the intercept anyway since you have to use Preact instead of React.

**Comment by Acmion** [comment]
A comparison with Preact could be interesting, however, the goal of this analysis was to determine whether Svelte is scalable. React is the most used framework and because it is considered scalable, then so is Svelte.

**Comment by lillallol** [comment]
> the goal of this analysis was to determine whether Svelte is scalable

Fine. The analysis might mislead framework consumers to always choose Svelte over React, while this should not always be the case. Not only that, people are using Preact to reduce the bundle size, which by the way decreases the inflection point by many kbs. This makes the analysis look biased in favor of Svelte.

I am personally looking at the analysis from a framework designer perspective. I do not use any of the major frameworks. I am trying to build my own. What are the gains of creating a new language (.svelte), creating IDE support for the language (syntax highlighting, refactoring, etc.), creating a compiler and forcing people to compile at the development stage? If you can still AOC from js code to js code, maybe creating a new language is not the optimal solution after all. If the js code has to be a certain way for it to optimally AOC, then create a linter, or give helpful error messages on compilation that will make the user understand how to change the code to be optimally AOC. 

Minimal time to interactive is good for SEO when a user is doing the first click to your web site, but after that, the inflection point forces you to send more bytes. So maybe creating a new language was a design mistake, since for that case you could send the js framework (which by the way you can AOC when you see fit). If the inflection point is something rare to hit, then fine, no problem. It is my understanding though that the inflection point is hit for almost every basic website when comparing Svelte with Preact. This is the reason why I am trying to calculate a better estimation of the inflection point.

**Comment by Acmion** [comment]
I agree that analyzing Preact would be valuable. Perhaps I will do this at one point.

I'd argue that the real benefits of Svelte are not in the bundle size, but rather in the following:
- The syntax. For example, I do not like how for loops (or maps) that generate components are implemented in JSX.
- Reactivity by default (no need to call some function to update the state).
- SvelteKit.
- Other stuff.
